### ISSUE TITLE
vcheck constantly asking for vcenter credentials

### ISSUE THREAD

**Issue by maZuFC** [alanrenouf/vCheck-vSphere#502]
hi all

im looking at deploying vcheck to all our regional offices which are spread around the world. So far i have managed to do 2 locations without too much trouble... but ive gotten stuck on the next office where i cannot for the life of me figure out why vcheck keeps asking for login credentials.

it doesnt matter if i run vcheck as administrator, as my admin account or as the vcheck service account it does not want to remember the login credentials so every time i re-run the script even if its in the same window session the credentials window pops up!
even logging onto the server as the vcheck service account and running everything from there doesnt seem to work.
i can confirm that my admin account and the vcheck service account have admin access within vcenter.

im not sure if ive missed a step but the setup process for this is quite easy so im a little stumped as to what is happening.

pre-requisites are all installed.
i have run the initial config setup and when i re-run it the details are still there so info is being saved.

where are the credentials meant to be stored?
am i missing a step?

any thoughts or suggestions will be greatly appreciated!

thanks

**Comment by klaus385** [comment]
It seems like plugin-41 manages this service and talks with plugin-00.

`If (! $VCSA) {
	If (Test-Path $Credfile) {
		$LoadedCredentials = Import-Clixml $Credfile
		$creds = New-Object System.Management.Automation.PsCredential($LoadedCredentials.Username,($LoadedCredentials.Password | ConvertTo-SecureString))
		$Services = get-wmiobject -Credential $creds win32_service -ComputerName $VIServer -ErrorAction SilentlyContinue| Where {$_.DisplayName -like "VMware*" }
	} Else {
		$Services = get-wmiobject win32_service -ComputerName $VIServer -ErrorAction SilentlyContinue | Where {$_.DisplayName -like "VMware*" }
		if ($Error[0].Exception.Message -match "Access is denied.") { 
			# Access Denied Error found so asking to store windows credentials in a file for future use
			Write-Host "Current windows credentials do not allow for access to WMI on the host $VIServer, please enter Administrator credentials for this check to work, these will be stored in an encrypted file: $credfile" 
			$Credential = Get-Credential
			$Pass = $credential.Password | ConvertFrom-SecureString
			$Username = $Credential.UserName
			$Store = "" | Select Username, Password
			$Store.Username = $Username
			$Store.Password = $Pass
			$Store | Export-Clixml $credfile
		$Services = get-wmiobject win32_service -ComputerName $VIserver -ErrorAction SilentlyContinue | Where {$_.DisplayName -like "VMware*" }
		}
	}
`
Its seems the Test-Path always fails thus prompts you every time you can hardcode it if needed. But, the problem seems to be in this location.

**Comment by maZuFC** [comment]
hmmm interesting... so where is this $credfile being written to? 
my powershell skills are quite limited so im not sure if im reading the script right...

sounding like a permissions issue to me at the moment :/

**Comment by Sneddo** [comment]
Sounds like your authentication isn't configured for pass through auth, which is why you get the prompt even running as an authorised account.

$credfile should reference a file called Windowscreds.xml in your vCheck directory. The catch with this as well, is that you must use the same account that creates the file to decrypt the contents.

Might be better to shift this to the credential store functionality (New-VICredentialStoreItem) at some point...

**Comment by maZuFC** [comment]
yeah you are right, there something wrong with the SSO in this location.
i cannot log into the vsphere client using the windows credentials option...

i can confirm that in the 2 locations that i have managed to rollout vcheck, the windows credentials option does work!

right so i think i need to resolve the SSO login feature for this option to work properly.


lets see how that goes...

**Comment by maZuFC** [comment]
ok so it seems like there is a authentication issue with SSO for the locations that are not working with vcheck.

to test I found another location where I was able to tick the windows credentials check box and roll out vcheck... this worked perfectly without any dramas.

 

 

I have followed a number of kb articles that show how to fix broken sso with the main ones being this...

 

Logging into vCenter Server 5.1 or 5.5 using the "Use Windows session credentials" option fails with the error: Cannot c…

Unable to log in to VMware vCenter Server Appliance with the Use Windows session credentials option in the vSphere Clien…

 

unfortunately this did not resolve the issue.

 

doing more digging around I found posts and blogs saying to check the network settings and hostname within the vcsa admin page.

so after logging in there is definitely something weird going on because the hostname is different to the hostname I'm connecting to.

 

eg..

 

vcenter01 is the hostname I'm connecting to

vcenter02 is the hostname in vcsa !

 

DNS entries only show vcenter01

there is no vcenter02

 

as a test I created a HOST (a record) DNS entry with vcenter02 and the ip number.. and surprise surprise I can now log in using the windows authentication check box.

 

 

so the question now is, why is the hostname different to the dns record... I wasn't the one who set this up but ill be the one who now has to clean it up. 

ill have to test now to see if this new DNS entry I just entered resolves the problem of vCheck constantly asking for credentials.

 

either way I think these locations and the dns entries have to be resolved. 

hopefully changing the hostname is easy and wont break anything... Id rather avoid having to rebuild vcenter from scratch ....
